apps/molgenis-components/src/components/forms/EditModal.vue

```diff
@@ -0,0 +1,254 @@
+<template>
+  <LayoutModal :title="title" :show="isModalShown" @close="handleClose">
+    <template #body>
+      <RowEdit
+        :id="id"
+        v-model="rowData"
+        :pkey="pkey"
+        :tableName="tableName"
+        :tableMetaData="tableMetaData"
+        :graphqlURL="graphqlURL"
+        :clone="clone"
+      >
+      </RowEdit>
+    </template>
+    <template #footer>
+      <RowEditFooter
+        class="modal-footer"
+        :id="id + '-footer'"
+        :tableName="tableName"
+        :errorMessage="errorMessage"
+        @cancel="handleClose"
+        @saveDraft="handleSaveDraftRequest"
+        @save="handleSaveRequest"
+      ></RowEditFooter>
+    </template>
+  </LayoutModal>
+</template>
+
+<script>
+import Client from "../../client/client.js";
+import LayoutModal from "../layout/LayoutModal.vue";
+import RowEditFooter from "./RowEditFooter.vue";
+import RowEdit from "./RowEdit.vue";
+
+export default {
+  name: "EditModal",
+  components: { LayoutModal, RowEditFooter, RowEdit },
+  data() {
+    return {
+      rowData: {},
+      tableMetaData: {},
+      client: null,
+      errorMessage: null,
+    };
+  },
+  props: {
+    id: {
+      type: String,
+      required: true,
+    },
+    tableName: {
+      type: String,
+      required: true,
+    },
+    isModalShown: {
+      type: Boolean,
+      required: true,
+    },
+    graphqlURL: {
+      type: String,
+      required: false,
+      default: () => "graphql",
+    },
+    pkey: {
+      type: Object,
+      required: false,
+      default: () => null,
+    },
+    clone: {
+      type: Boolean,
+      required: false,
+      default: () => false,
+    },
+    visibleColumns: {
+      type: Array,
+      required: false,
+      default: () => [],
+    },
+    defaultValue: {
+      type: Object,
+      required: false,
+      default: () => null,
+    },
+  },
+  computed: {
+    title() {
+      return this.titlePrefix + " " + this.tableName;
+    },
+    titlePrefix() {
+      return this.pkey && this.clone ? "copy" : this.pkey ? "update" : "insert";
+    },
```

Comment:
As far as I see this is only used here, can this then better not just be a const inside the title()?